cmd/metal-api/internal/grpc/waitServer.go

```diff
@@ -247,7 +247,9 @@ func (s *WaitServer) Wait(req *v1.WaitRequest, srv v1.Wait_WaitServer) error {
 	defer func() {
 		s.queueLock.Lock()
 		delete(s.queue, machineID)
-		close(can)
+		if err != nil {
```

Comment:
cant immediately see where this error is coming from

Reply:
This happened a few times during 100+ subsequent test runs

Reply:
I'll revert this for now and try to test such a scenario

Reply:
OK, now the tests fail, this is weird. so we need the errcheck